internal/observability/observability.go

```diff
@@ -88,3 +91,18 @@ func exporter() traceAndViewExporter {
 		return sde
 	}
 }
+
+// ExposeMetricsIfPrometheus adds the /metrics endpoint to a mux if the correct env var is set
+func ExposeMetricsIfPrometheus(mux *http.ServeMux) error {
+	exporter := os.Getenv("OBSERVABILITY_EXPORTER")
+	if strings.EqualFold(exporter, "prometheus") {
+		exporter, err := prometheus.NewExporter(prometheus.Options{})
+		if err != nil {
+			return fmt.Errorf("failed to create prometheus exporter: %v", err)
+		}
+
+		mux.Handle("/metrics", exporter)
```

Comment:
how is access controlled?

Reply:
Seems like we may need to spin up a separate server in a go process so we can serve this on a different port.

Reply:
either that or front these servers w/ some kind of proxy that can apply ACLs based on path.

on cloud run, we can only listen on a single port, but also wouldn't make sense to enable prometheus in that environment anyway.

Reply:
I had been thinking of using the kube-rbac-proxy project but am now thinking a different port would be nicer.

Reply:
@mikehelmick I refactored as mentioned above. Could use another look when you get a chance.

Reply:
thanks! looking now